Assets/Plugins/UniRx/Scripts/Operators/ObserveOn.cs

```diff
@@ -47,89 +68,86 @@ public IDisposable Run()
 
                 return StableCompositeDisposable.Create(sourceDisposable, Disposable.Create(() =>
                 {
-                    lock (scheduleDisposables)
+                    lock (actions)
                     {
                         isDisposed = true;
 
-                        foreach (var item in scheduleDisposables)
-                        {
-                            item.Dispose();
-                        }
-                        scheduleDisposables.Clear();
+                        while (actions.Count > 0)
+                            actions.First.Value.Dispose();
```

Comment:
this requires a bit of parsing to see that it's correct. I would suggest a comment here indicating that Dispose actually removes the node from the list.